lib/Builtins.h

```diff
@@ -38,8 +39,9 @@ struct ParamTypeInfo {
   bool is_signed = false;                            // is element type signed
   llvm::Type::TypeID type_id = llvm::Type::VoidTyID; // element type
   uint32_t byte_len = 0;                             // element byte length
-  int vector_size = 0; // number of elements (0 == not a vector)
-  std::string name;    // struct name
+  int vector_size = 0;          // number of elements (0 == not a vector)
+  std::string name;             // struct name
+  clspv::AddressSpace::Type AS; // AddressSpace if type_id is PointerTyID
```

Comment:
Where does this get assigned? I can only see default initialization. In the cpp file I only see rvalue uses. It seems like it should be parsed.

Reply:
This gets assigned in `lib/LowerAddrSpaceCastPass.cpp:208`.
It is only used in `Builtins::GetMangledFunctionName` where I added the case `Type::PointerTyID`. This only function setting an arg info as `PointerTyID` is in `LowerAddrSpaceCastPass.cpp`.

Reply:
I went with another way of dealing with this. I used `Builtins::GetMangledFunctionName(const char *, Type *)` which already supports `PointerTyID`.